internal/services/authorization/role_management_policy.go

```diff
@@ -267,7 +267,9 @@ func buildRoleManagementPolicyForUpdate(metadata *sdk.ResourceMetaData, rolePoli
 					}
 				} else {
 					if approvalStagesRaw, ok := settings["approvalStages"]; ok {
-						approvalStages = approvalStagesRaw.([]map[string]interface{})
+						for _, stage := range approvalStagesRaw.([]interface{}) {
+							approvalStages = append(approvalStages, stage.(map[string]interface{}))
```

Comment:
We should check that `stage` is a `map[string]interface{}` before casting to it to prevent another potential crash